src/page/CartPage/index.jsx

```diff
@@ -0,0 +1,116 @@
+// @ts-nocheck
+import { useCallback, useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+
+import { Image, CartProductItem, CheckBox, TotalPrice } from 'components';
+
+import store from 'store/store';
+import {
+  doAddProductToOrder,
+  doInitializeOrder,
+  doSelectiveDeleteFromCart,
+} from 'actions/actionCreator';
+import empty from 'assets/empty.jpeg';
+import Styled from 'page/CartPage/index.style';
+import { useNavigate } from 'react-router-dom';
+import useSnackbar from 'hooks/useSnackbar';
+import { MESSAGE } from 'utils/constants';
+
+const CartPage = () => {
+  const [renderSnackbar] = useSnackbar();
+  const navigate = useNavigate();
+  const { isAuthenticated } = useSelector(state => state.authReducer);
+  const { products, shoppingCart, order } = useSelector(state => state.reducer);
+  const [totalPrice, setTotalPrice] = useState(0);
+
+  const calculateTotalPrice = useCallback(() => {
+    let total = 0;
+
+    order.forEach(productId => {
+      const { price } = products.find(product => product.id === productId);
+      const { quantity } = shoppingCart.find(product => product.id === productId);
+
+      total += quantity * price;
+    });
+
+    return total;
+  }, [products, shoppingCart, order]);
+
+  useEffect(() => {
+    if (!isAuthenticated) {
+      renderSnackbar(MESSAGE.NO_AUTHORIZATION, 'FAILED');
+      navigate('/login');
+    }
+    setTotalPrice(calculateTotalPrice());
+  }, []);
```

Comment:
이미 merge가 돼서, 새로고침 시 로그인 유지 불가 에러에 대해 수정한 코드를 코멘트로 적어두겠습니다.
```suggestion
  useEffect(() => {
    if (isLoading) return;
    if (!isAuthenticated) {
      renderSnackbar(MESSAGE.NO_AUTHORIZATION, 'FAILED');
      navigate('/login');
    }
    setTotalPrice(calculateTotalPrice());
  }, [calculateTotalPrice, isAuthenticated, isLoading, navigate, renderSnackbar]);
```